crypto/src/main/java/org/cryptimeleon/incentive/crypto/IncentiveSystem.java

```diff
@@ -440,17 +442,21 @@ Hence, the provider can easily retrieve usk and esk (using the Schnorr-trick, co
      * React to a legitimate spend request to allow the user retrieving an updated token with the value decreased by k.
      * Returns additional data for double-spending protection.
      *
-     * @param spendRequest    the user's request
-     * @param providerKeyPair keypair of the provider
-     * @param tid             transaction id, should be verified by the provider
-     * @param spendDeductTree the zero knowledge proof to verify for this promotion
+     * @param spendRequest          the user's request
+     * @param promotionParameters   specifying the promotion that the user wants to spend his points on (i.e. her claim)
+     * @param providerKeyPair       keypair of the provider
+     * @param tid                   transaction id, should be verified by the provider
+     * @param spendDeductTree       the zero knowledge proof to verify for this promotion
+     * @param userChoice            byte representation of the user choice,
+     *                              influences the computation of challenge generator gamma and provider share esk_prov
      * @return tuple of response to send to the user and information required for double-spending protection
      */
     public DeductOutput generateSpendRequestResponse(PromotionParameters promotionParameters,
                                                      SpendRequest spendRequest,
                                                      ProviderKeyPair providerKeyPair,
                                                      ZnElement tid,
-                                                     SpendDeductTree spendDeductTree) {
+                                                     SpendDeductTree spendDeductTree,
+                                                     UniqueByteRepresentable userChoice) {
```

Comment:
**Idea:** Why don't you overload this method (and the other two Spend methods) to put tid in place of userChoice and mark the overloaded method as deprecated. Then we do not need to artificially put tid in place of userChoice everywhere this method is used, and we can use this api once we decided upon what userChoice is.

Reply:
Would have been a possible way to quickfix this, for sure. However, a downside is us forgetting to remove such a deprecated spend-algo call. So we would need a flashing todo next to each of these calls (as right now) anyway.